contracts/andromeda_mission/src/state.rs

```diff
@@ -0,0 +1,69 @@
+use andromeda_protocol::mission::MissionComponent;
+use common::{ado_base::AndromedaMsg, error::ContractError};
+use cosmwasm_std::{to_binary, Addr, Coin, CosmosMsg, Order, ReplyOn, Storage, SubMsg, WasmMsg};
+use cw_storage_plus::{Bound, Item, Map};
+
+/// Used to store the addresses of each ADO within the mission
+pub const ADO_ADDRESSES: Map<&str, Addr> = Map::new("ado_addresses");
+/// Stores a record of the describing structs for each ADO
+pub const ADO_DESCRIPTORS: Map<&str, MissionComponent> = Map::new("ado_descriptors");
+pub const ADO_IDX: Item<u64> = Item::new("ado_idx");
+pub const MISSION_NAME: Item<String> = Item::new("mission_name");
+
+// DEV NOTE: Very similar to CW721 module instantiation, possibly merge both implementations?
+pub fn add_mission_component(
+    storage: &mut dyn Storage,
+    component: &MissionComponent,
+) -> Result<u64, ContractError> {
+    let idx = match ADO_IDX.load(storage) {
+        Ok(index) => index,
+        Err(..) => 1u64,
+    };
```

Comment:
nit: This can be simplified to `let idx = ADO_IDX.may_load(storage)?.unwrap_or(1u64)`.